avocado_vt/plugins/vt_joblock.py

```diff
@@ -86,35 +87,17 @@ def _get_lock_files(self):
             if e.errno == errno.ENOENT:
                 return []
 
-    def _get_lock_file_pid(self):
-        """
-        Gets the lock file path and the process ID
-
-        If the lock file can not be located, this returns (None, 0).
-
-        :returns: the path and the (integer) process id
-        :rtype: tuple(str, int)
-        """
-        files = self._get_lock_files()
-        if not files:
-            return (None, 0)
-        path = files[0]
-        content = int(open(path, 'r').read())
-        pid = int(content)
-        if pid > 0:
-            return (path, pid)
-
     def _lock(self, job):
-        filename, lock_pid = self._get_lock_file_pid()
-        if lock_pid > 0:
-            msg = ('Avocado-VT job lock file "%s" acquired by PID %u. '
-                   'Aborting...' % (filename, lock_pid))
-            self._abort(msg, job)
         self.lock_file = self._create_self_lock_file(job)
-
-    def _unlock(self):
-        if self.lock_file:
-            os.unlink(self.lock_file)
+        lock_files = self._get_lock_files()
+        lock_files.remove(self.lock_file)
+        for path in lock_files:
+            lock_pid = int(open(path, 'r').read())
+            if pid_exists(lock_pid):
+                msg = 'File "%s" acquired by PID %u. ' % (path, lock_pid)
+                raise OtherProcessHoldsLockError(msg)
+            else:
+                os.unlink(path)
```

Comment:
I thought this was also answered by a previous comment... the way I feel about `_lock()` is that if *anything* wrong happens, that is, *any* exception is raised, it will be handled by the `try/except` block in `pre()`, effectively bailing out of the job and exiting Avocado.  Now, with the logging of the traceback, as per your suggestion, it becomes even clearer what caused the failure, such as the lack of permission. Example:

```
$ avocado run boot 
Failure trying to set Avocado-VT job lock: [Errno 1] Operation not permitted: '/tmp/avocado-vt-joblock-6faebbe2cd4e144fa7bd8d85ad68336c99af971f-1000-424mzl58.pid'

Reproduced traceback from: /home/cleber/src/avocado/avocado-vt/avocado_vt/plugins/vt_joblock.py:117
Traceback (most recent call last):
  File "/home/cleber/src/avocado/avocado-vt/avocado_vt/plugins/vt_joblock.py", line 107, in _lock
    os.unlink(path)
OSError: [Errno 1] Operation not permitted: '/tmp/avocado-vt-joblock-6faebbe2cd4e144fa7bd8d85ad68336c99af971f-1000-424mzl58.pid'
```

Do you see something wrong with that approach?